src/dharma.ts

```diff
@@ -58,6 +74,24 @@ class Dharma {
         this.blockchain = new BlockchainAPI(this.web3, this.contracts);
         this.logs = new LogsAPI(this.web3, this.contracts);
     }
+
+    private verifyWeb3Version(web3: Web3) {
```

Comment:
this function smells funky to me. is our version requirements so strict that it needs to be the exact version? and do we not have a minimum version and max version that we can just specify as constants somewhere.